### PR TITLE
#5701 Optimize way of adding multiple jars to classpath.

### PR DESCRIPTION
**Pull request by michalgce**

### CODE REVIEW

**Comment by michalgce** [comment]
Merge firstly #5651

File: kernel/base/src/main/java/com/twosigma/beakerx/evaluator/BaseEvaluator.java [hunk 1]
```diff
@@ -27,10 +32,27 @@
...
+  @Override
+  public List<Path> addJarsToClasspath(List<PathToJar> paths) {
+    LinkedList<Path> addedPaths = Lists.newLinkedList();
+    paths.forEach(path -> {
+      if (addJar(path)) {
+        addedPaths.add(Paths.get(path.getPath()));
+      }
+    });
+
+    resetEnvironment();
```
Comment (jpallas) [inline kernel/base/src/main/java/com/twosigma/beakerx/evaluator/BaseEvaluator.java:53]: Should this be conditional on `!addedPaths.isEmpty()` for consistency with the single jar case?

Comment (michalgce) [reply kernel/base/src/main/java/com/twosigma/beakerx/evaluator/BaseEvaluator.java:53]: Good Point!

**Comment by michalgce** [comment]
I closed this PR and open another which comes from master instead of #5651

**Pull request closed without merging**